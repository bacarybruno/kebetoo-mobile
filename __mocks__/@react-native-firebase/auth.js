import authors from '@fixtures/authors';

const user = {
  ...authors[0],
  updateProfile: jest.fn(),
};

export const authCredentials = {
  user,
  additionalUserInfo: {
    profile: { id: 'mock-user-profile-id' },
  },
};

const mockOptions = {
  createUserAndRetrieveDataWithEmailAndPassword: jest.fn().mockReturnValue(Promise.resolve(true)),
  sendPasswordResetEmail: jest.fn().mockReturnValue(Promise.resolve(true)),
  signInAndRetrieveDataWithEmailAndPassword: jest.fn().mockReturnValue(Promise.resolve(true)),
  fetchSignInMethodsForEmail: jest.fn().mockReturnValue(Promise.resolve(true)),
  signOut: jest.fn().mockReturnValue(Promise.resolve(true)),
  onAuthStateChanged: jest.fn(),
  signInWithCredential: jest.fn().mockReturnValue(authCredentials),
  credentials: authCredentials,
  signInWithEmailAndPassword: jest.fn().mockReturnValue(Promise.resolve({ user })),
  createUserWithEmailAndPassword: jest.fn().mockReturnValue(Promise.resolve({ user })),
  currentUser: user,
};

const auth = jest.fn(() => mockOptions);

const socialAuthProviderData = {
  credential: jest.fn().mockReturnValue({
    providerId: 'jest',
    secret: 'mock-google-credentials-secret',
    token: 'mock-google-credentials-token',
  }),
};

auth.GoogleAuthProvider = socialAuthProviderData;

auth.FacebookAuthProvider = socialAuthProviderData;

auth.setMockOptions = (options) => {
  Object.keys(options).forEach((key) => {
    mockOptions[key] = options[key];
  });
};

export default auth;
