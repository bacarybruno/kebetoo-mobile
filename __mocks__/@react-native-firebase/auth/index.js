const user = {
  email: 'mockEmail',
  uid: 'mockUid',
  displayName: 'mockDisplayName',
  photoURL: 'https://avatars1.githubusercontent.com/u/14147533',
  updateProfile: jest.fn(),
}

export const authCredentials = {
  user,
  additionalUserInfo: {
    profile: { id: 'mock-user-profile-id' },
  },
}

const mockOptions = {
  createUserAndRetrieveDataWithEmailAndPassword: jest.fn().mockReturnValue(Promise.resolve(true)),
  sendPasswordResetEmail: jest.fn().mockReturnValue(Promise.resolve(true)),
  signInAndRetrieveDataWithEmailAndPassword: jest.fn().mockReturnValue(Promise.resolve(true)),
  fetchSignInMethodsForEmail: jest.fn().mockReturnValue(Promise.resolve(true)),
  signOut: jest.fn().mockReturnValue(Promise.resolve(true)),
  onAuthStateChanged: jest.fn(),
  signInWithCredential: jest.fn().mockReturnValue(authCredentials),
  credentials: authCredentials,
  currentUser: user,
}

const auth = jest.fn(() => mockOptions)

const socialAuthProviderData = {
  credential: jest.fn().mockReturnValue({
    providerId: 'jest',
    secret: 'mock-google-credentials-secret',
    token: 'mock-google-credentials-token',
  }),
}

auth.GoogleAuthProvider = socialAuthProviderData

auth.FacebookAuthProvider = socialAuthProviderData

auth.setMockOptions = (options) => {
  Object.keys(options).forEach((key) => {
    mockOptions[key] = options[key]
  })
}

export default auth
