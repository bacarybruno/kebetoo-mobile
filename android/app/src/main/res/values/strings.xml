<resources>
    <string name="app_name">Kebetoo</string>
</resources>
