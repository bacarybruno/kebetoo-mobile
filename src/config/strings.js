import LocalizedStrings from 'react-native-localization';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import 'dayjs/locale/en';
import 'dayjs/locale/fr';

import frStrings from '@assets/locales/fr';
import enStrings from '@assets/locales/en';

const strings = new LocalizedStrings({
  fr: frStrings,
  en: enStrings,
});

dayjs.locale(strings.getLanguage());
dayjs.extend(relativeTime);

export const updateAppLocale = (locale) => {
  dayjs.locale(locale);
  strings.setLanguage(locale);
};

export default strings;
