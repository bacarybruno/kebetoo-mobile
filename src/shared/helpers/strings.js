export const capitalize = (s) => {
  if (typeof s !== 'string') return ''
  const splitCharacters = '. '
  const parts = s.split(splitCharacters)
  return parts
    .map((s) => s.charAt(0).toUpperCase() + s.slice(1).toLowerCase())
    .join(splitCharacters)
}
