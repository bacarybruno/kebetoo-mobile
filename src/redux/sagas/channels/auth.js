import { take, call, put } from 'redux-saga/effects'
import { eventChannel } from 'redux-saga'
import auth from '@react-native-firebase/auth'

import * as types from '../../types'

function getAuthChannel() {
  if (!this.authChannel) {
    this.authChannel = eventChannel((emit) => {
      const unsubscribe = auth().onUserChanged((user) => {
        let payload = { isLoggedIn: false }

        if (user) {
          payload = {
            ...payload,
            uid: user.uid,
            isLoggedIn: true,
            email: user.email,
            photoURL: user.photoURL,
            displayName: user.displayName,
            providerUid: user.providerData[0].uid,
            provider: user.providerData[0].providerId,
          }
        }

        if (payload.provider === 'password' && !payload.displayName) {
          // user have been created with email and password
          // but displayName is not yet assigned
          // we wait for the next event that will be assigning displayName to user
          return
        }

        emit({ payload })
      })
      return unsubscribe
    })
  }
  return this.authChannel
}

function* watchForFirebaseAuth() {
  const authChannel = yield call(getAuthChannel)
  try {
    while (true) {
      const { payload } = yield take(authChannel)
      console.log('Received auth event', payload)
      yield put({ type: types.SET_USER_PROFILE_REQUEST, payload })
    }
  } catch (error) {
    console.log('An error occured in auth channel', error)
  } finally {
    console.log('Auth channel terminated gracefully')
  }
}

export default watchForFirebaseAuth
