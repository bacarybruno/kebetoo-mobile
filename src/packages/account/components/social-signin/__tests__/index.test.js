import React from 'react'
import { fireEvent } from 'react-native-testing-library'
import { act } from 'react-test-renderer'
import auth from '@react-native-firebase/auth'

import setupTest from 'Kebetoo/src/config/jest-setup'
import strings from 'Kebetoo/src/config/strings'
import Text from 'Kebetoo/src/shared/components/text'

import SocialSignin from '../index'

const givenSocialSignin = setupTest(SocialSignin)({
  text: strings.auth.or_signin_with,
  onSignIn: jest.fn(),
  children: <Text text={strings.auth.dont_have_account} />,
})

it('renders SocialSignin', () => {
  const { wrapper } = givenSocialSignin()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

describe('sign in', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const onSignInPayload = expect.objectContaining({
    data: auth().credentials,
    error: null,
  })

  const testProvider = async (testID) => {
    const { wrapper, props } = givenSocialSignin()
    const signinButton = wrapper.root.findByProps({ testID })
    await act(async () => {
      await fireEvent.press(signinButton)
    })
    return props.onSignIn
  }

  test('with google', async () => {
    const onSignIn = await testProvider('google-signin')
    expect(onSignIn).toBeCalledTimes(1)
    expect(onSignIn).toBeCalledWith(onSignInPayload)
  })

  test('with facebook', async () => {
    const onSignIn = await testProvider('facebook-signin')
    expect(onSignIn).toBeCalledTimes(1)
    expect(onSignIn).toBeCalledWith(onSignInPayload)
  })
})
