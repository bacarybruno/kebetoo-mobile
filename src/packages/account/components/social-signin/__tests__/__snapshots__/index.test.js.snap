// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SocialSignin 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "justifyContent": "space-evenly",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "#707070",
              "borderWidth": 0.5,
              "flex": 1,
              "height": 0.5,
            }
          }
        />
        <View
          style={
            Object {
              "borderColor": "#707070",
              "borderWidth": 0.5,
              "flex": 1,
              "height": 0.5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="facebook-signin"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#3b5998",
                  "fontSize": 33,
                },
                undefined,
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î¤…
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="google-signin"
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/google.png",
              }
            }
            style={
              Object {
                "height": 33,
                "marginHorizontal": 8,
                "width": 32,
              }
            }
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Helvetica",
          },
          Object {
            "color": "rgba(0, 0, 0, 0.8)",
            "fontSize": 16,
            "fontWeight": "normal",
            "opacity": 1,
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      Don't have an account?
    </Text>
  </View>
</View>
`;
